### ISSUE TITLE
Changing locales after deleting search engines only restores partial list of search engines

### ISSUE THREAD

**Issue by npark-mozilla** [mozilla-mobile/focus-android#1897]
STR:
- In English, go to Installed search engines dialog, and delete all search engines except the default one
- exit, and go to language setting. 
- Select another locale, like Korean where they have a different set of default search engines.
- Go back to installed search engines dialog.

Expected:
- Either all search engines are restored, or only the default search engine remains
Actual:
- Partial list of search engines are restored.

Workaround: tap 'restore default search engines' again.

Minor issue, but still worth to report it.

**Comment by mcomella** [comment]
No-Jun mentions this might be tied to something more sinister: make sure this isn't hiding any big code errors.

**Comment by tvalle** [comment]
Can I look into this?

I think that the expected behaviour is all search engines being restored since the user can select a search engine that is not available in other languages

**Comment by mcomella** [comment]
@tvalle Go for it! Sorry for the delay, we've been on vacation for the holidays.

I think what's important here:
- to preserve the selected default search engine in each locale (e.g. for a user that switches locales often) 
- to keep any custom search engines the user has added (which should probably be available in all locales).

For this bug, I think ideally each locale will keep it's own set of search engines so:
- In english, remove all but Google
- Switch to Korean, all search engines shown (including google if relevant)
- Remove all but Naver (or whatever)
- Switch to english, only Google shown
- Switch to Korean, only Korean shown

But if that's too much code (*please be very aggressive about this*, it's really not worth adding a lot of code for non-crashy edge cases), I wouldn't mind just restoring a locale's search engines on locale change.

**Comment by levimatheri** [comment]
@npark-mozilla When you switch to Korean, do you get search engines specific to Korea? I don't get that when I try it

**Comment by levimatheri** [comment]
@mcomella I have a fix for it, I just need to write a test for it then make a PR.
 The only thing is when I switch to another language, the search engines are restored, but it doesn't restore specific to the locale. This might be an issue to raise since it happens with the release version as well

**Comment by levimatheri** [comment]
@mcomella So I just tried clearing cache for the app and it restores the search engines correctly (specific to the locale in question). So the issue is it should pull up the locale's search engines as soon as the locale is changed
